### PR TITLE
Revise MatrixGroup and use it everywhere

### PR DESCRIPTION
**Pull request by maxhauck**
- Replace all uses of `Group` by `MatrixGroup` or `MatrixGroupWithSize`, respectively.
- Add tests for `MatrixGroup` and `MatrixGroupWithSize`

### General
- [ ] All functions have unit tests

### Functions constructing generators of maximal subgroups
- [ ] The code which computes and then stores the sizes is correct. To do this one confers to the tables referenced by the code. Tests for the group size should use `RECOG.TestGroup` from the recog package.
- [ ] The test suite checks whether all constructed generators are sensible. That is it tests the generators'
  - [ ] determinants (and if applicable also the spinor norms are correct), and
  - [ ] compatibility with the correct form,
  - [ ] `DefaultFieldOfMatrixGroup` returns the correct field.

### Functions assembling the list of all maximal subgroups of a certain group
- [ ] The code agrees with the tables in section 8.2 of the book "The Maximal Subgroups of the Low-dimensional Finite Classical Groups".

The reviewer doesn't need to compare our results to magma's results. That's the job of the person implementing the code.

### CODE REVIEW

File: gap/ClassicalNormalizerMatrixGroups.gi
Comment (ssiccha) [inline gap/ClassicalNormalizerMatrixGroups.gi:80]: We don't need `result` anymore, right?

File: gap/ClassicalNormalizerMatrixGroups.gi
Comment (ssiccha) [inline gap/ClassicalNormalizerMatrixGroups.gi:19]: This can be moved to where `C` is defined.

File: gap/Utils.gd
Comment (fingolfin) [inline gap/Utils.gd:31]: This is wrong, even if `F` is finite: it (currently) only holds for fields with <= 256 elements, as then `ImmutableMatrix` will convert the matrices into "compressed" matrices which store the field explicitly. Indeed:
```
gap> F:=GF(257^2);
GF(257^2)
gap> G:=MatrixGroup(F, [IdentityMat(2,F)]);
Group([ [ [ Z(257)^0, 0*Z(257) ], [ 0*Z(257), Z(257)^0 ] ] ])
gap> DefaultFieldOfMatrixGroup(G);
GF(257)
```

Comment (fingolfin) [reply gap/Utils.gd]: OK, I see you changed the code since I wrote this, so I guess the comment above is moot...

File: gap/ClassicalNormalizerMatrixGroups.gi
Comment (ssiccha) [reply gap/ClassicalNormalizerMatrixGroups.gi:70]: Hmm, I never thought about it that way. I actually like returning the result of this function more than "define result, then call `SetSize`, then return".

File: gap/Utils.gi
Comment (ssiccha) [reply gap/Utils.gi]: But I think that, until then this can be merged such that everyone can start using the different argument order of MatrixGroup.

**Review by ssiccha — APPROVED** [review]

**Comment by ssiccha** [comment]
I can either merge this after lunch and we open a separate PR to make true on the promise for `DefaultFieldOfMatrixGroup`, or we wait until everything is resolved and merge then. I'd do the first option.

**Pull request merged**
